src/CargoBuild/sdk/Sdk.targets

```diff
@@ -0,0 +1,130 @@
+	<Target Name="GetReferenceAssemblyPaths" DependsOnTargets="$(GetReferenceAssemblyPathsDependsOn)" />
+
+	<Import Project="$(CustomAfterNoTargets)" Condition="'$(CustomAfterNoTargets)' != '' and Exists('$(CustomAfterNoTargets)')" />
+
+	<!-- 
+    Microsoft.Managed.Targets is imported by the managed language target files in MSBuild 16.0 and above, but most of the msbuild tasks are actually in Microsoft.Common.Currentversion.targets.
+    So import it when the managed targets do not get imported.
+  -->
+	<Import Project="$(MSBuildToolsPath)\Microsoft.Managed.targets" Condition="'$(MSBuildAssemblyVersion)' >= '16.0' And '$(ManagedLanguageTargetsGotImported)' != 'true'" />
+
+	<!-- Override stock CoreCompile target to do nothing but keep extensibility points -->
+	<Target Name="CoreCompile"
+			DependsOnTargets="$(CoreCompileDependsOn)">
+		<CallTarget Targets="$(TargetsTriggeredByCompilation)" Condition="'$(TargetsTriggeredByCompilation)' != ''" />
+	</Target>
+
+	<Target Name="_GenerateCompileInputs" />
+	<Target Name="_GenerateCompileDependencyCache" />
+
+	<!-- Disables the _CopyFilesMarkedCopyLocal target to not copy references when SkipCopyFilesMarkedCopyLocal is set to true. -->
+	<Import Project="DisableCopyFilesMarkedCopyLocal.targets" Condition="'$(SkipCopyFilesMarkedCopyLocal)' == 'true'" />
+	<!-- Rust specific targets -->
+	<UsingTask TaskName="MSBuild.CargoBuild.CargoBuildTask" AssemblyFile="$(MSBuildThisFileDirectory)..\build\net6.0\Microsoft.Build.CargoBuild.dll" Condition="'$(MSBuildRuntimeType)' == 'Core'"/>
+	<UsingTask TaskName="MSBuild.CargoBuild.CargoBuildTask" AssemblyFile="$(MSBuildThisFileDirectory)..\build\net472\Microsoft.Build.CargoBuild.dll" Condition="'$(MSBuildRuntimeType)' != 'Core'"/>
+	<Target Name="CargoFetch" AfterTargets="Restore">
+		<CargoBuildTask EnableAuth="False" StartupProj="$(MSBuildProjectFullPath)" RegistryFeedName="$(CargoRegistryFeedName)" Command="fetch" />
+	</Target>
+	<Target Name="CargoBuild" AfterTargets="CoreCompile">
+		<CargoBuildTask EnableAuth="False" StartupProj="$(MSBuildProjectFullPath)" Command="build" CommandArgs="$(BuildCommandArgs)" />
+	</Target>
+	<Target Name="Test" AfterTargets="CargoBuild" Condition="$(Test)=='true'">
+		<CargoBuildTask StartupProj="$(MSBuildProjectFullPath)" Command="test" CommandArgs="$(TestCommandArgs)" />
+	</Target>
+	<Target Name="Clean" BeforeTargets="CargoBuild" Condition="$(Clean)=='true'">
+		<CargoBuildTask StartupProj="$(MSBuildProjectFullPath)" Command="clean" CommandArgs="$(CleanCommandArgs)" />
+	</Target>
+	<Target Name="Run" AfterTargets="CargoBuild" Condition="$(Run)=='true'">
+		<CargoBuildTask StartupProj="$(MSBuildProjectFullPath)" Command="run" CommandArgs="$(RunCommandArgs)" />
+	</Target>
+	<Target Name="Doc" AfterTargets="CargoBuild" Condition="$(Doc)=='true'">
+		<CargoBuildTask StartupProj="$(MSBuildProjectFullPath)" Command="doc" CommandArgs="$(DocCommandArgs)" />
+	</Target>
```

Comment:
For these conditions, where do these properties come from? Why are there conditions at all?

Shouldn't the `AfterTargets` be a `DependsOnTargets`?